app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/uploads/view/CourseUploadsFragment.kt

```diff
@@ -73,7 +73,11 @@ class CourseUploadsFragment : Fragment(), CourseUploadsView {
         if (type == 1) {
             courseUrl?.let { courseUploadsPresenter.requestCourseUploads(it) }
         } else if (type == 2) {
-            courseUploadList?.let { setData(it) }
+            courseUploadList?.let { setData(it)
+            } ?:run {
```

Comment:
AFAIK run is also doing null check. We need to check if size of list is 0 and change the visibility accordingly. Please correct me if I am wrong.